### PR TITLE
Devs

### PR DESCRIPTION
**Pull request by Dzionek95**
Hello!

This is fully playable game without one feature which is Nuke

### CODE REVIEW

File: src/main/java/com/java_academy/Main.java [hunk 1]
```diff
@@ -18,6 +20,7 @@ public void start(Stage primaryStage) throws Exception {
 
 
     public static void main(String[] args) {
+    	I18NResolver.getI18NResolverInstance();
```
Comment (ghost) [inline src/main/java/com/java_academy/Main.java:23]: name of this ?

File: src/main/java/com/java_academy/gui/Controller.java [hunk 2]
```diff
@@ -50,25 +54,40 @@
...
-                               view.drawMiss((Pane)n);
-                           }
-                       }
-                   }
-               }
+        if(isMy) {
+            for (Node n : gridPaneShips.getChildren()) {
+                if (n instanceof Pane) {
+                    for(Map.Entry<Integer, Boolean> entry: board.entrySet()){
+                        if((new Integer(entry.getKey() + 100)).equals(transformationOfSourceIntoInteger(((Pane)n).getId()))) {
```
Comment (stabri) [inline src/main/java/com/java_academy/gui/Controller.java:61]: Bug here. FindBugs

File: src/main/java/com/java_academy/gui/Controller.java [hunk 3]
```diff
@@ -50,25 +54,40 @@
...
+                                view.drawShot((Pane) n);
+                            } else {
+                                view.drawMiss((Pane) n);
+
+                                setButtonsDisabled(true);
+                                connector.sendMessage(new MessageObject(null, "to stanPosredni"));
+                            }
+                        }
+                    }
+                }
```
Comment (ghost) [inline src/main/java/com/java_academy/gui/Controller.java:90]: to refactor

File: src/main/java/com/java_academy/gui/Controller.java [hunk 4]
```diff
@@ -81,7 +100,8 @@ public void showNuke() {
     public void onShootHandled(MouseEvent event) {
         Object source = event.getSource();
         Integer id = transformationOfSourceIntoInteger(source);
-//        System.out.println("Kliknalem id: " + id);
+        connector.sendMessage(new MessageObject(null, ""+id));
```
Comment (ghost) [inline src/main/java/com/java_academy/gui/Controller.java:103]: String.valueOf(id);

File: src/main/java/com/java_academy/logic/state_machine/SwitchBlockingBoardState.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    public void display(Consumer<MessageObject> displayConsumer) {
+        displayConsumer.accept(new MessageObject(currentPlayer, MessageCreator.createJsonMessageByKey("your.turn")));
+        displayConsumer.accept(new MessageObject(currentPlayer.getOpponent(), MessageCreator.createJsonMessageByKey("not.your.turn")));
+    }
+
+    @Override
+    public GameState changeState(String message) {
+        return new PlayerActionState(currentPlayer);
+    }
+}
```
Comment (ghost) [inline src/main/java/com/java_academy/logic/state_machine/SwitchBlockingBoardState.java:31]: clean :+1:

File: src/main/java/com/java_academy/network/socket_provider/ClientSocketProvider.java [hunk 6]
```diff
@@ -44,11 +44,8 @@ public boolean connect(InetSocketAddress address) {
 
     @Override
     public void sendMessage(MessageObject messageObject) {
-        try {
-            outputDataProcessor.sendMessage(messageObject.getMessage());
-        } catch (NullPointerException e){
-            BSLog.warn(LOGGER, I18NResolver.getMsgByKey("NULL_POINTER_HANDLE_MESSAGE", e.getMessage()));
-        }
+        messageQueue.add(messageObject);
+        sendNextMessage();
```
Comment (ghost) [inline src/main/java/com/java_academy/network/socket_provider/ClientSocketProvider.java:48]: clean :+1:

File: src/main/java/com/java_academy/network/socket_provider/core/OnMessageSentListener.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * @author Siarhei Shauchenka
+ * @since 04.08.17
+ */
+@FunctionalInterface
+public interface OnMessageSentListener {
+
+    void onMessageSent();
+}
```
Comment (ghost) [inline src/main/java/com/java_academy/network/socket_provider/core/OnMessageSentListener.java:11]: :1st_place_medal:

**Review by ghost — APPROVED** [review]

File: src/main/java/com/java_academy/gui/Controller.java (hunk 4 above)
Comment (ghost) [reply src/main/java/com/java_academy/gui/Controller.java:103]: Integer.toString(id);

**Review by ghost — APPROVED** [review]

**Review by stabri — APPROVED** [review]

**Pull request merged**